docs/USAGE-GUIDE.md

```diff
@@ -1897,7 +1903,13 @@ to either **S2N_BLOCKED_ON_READ** or **S2N_BLOCKED_ON_WRITE**, depending on the
 direction in which s2n-tls is blocked.
 
 s2n-tls I/O functions should be called repeatedly until the **blocked** parameter is
-**S2N_NOT_BLOCKED**. 
+**S2N_NOT_BLOCKED**.
+
+Applications should be designed so that one process is reading from a pipe before another
+writes to it. If the reading process fails to start or is terminated unexpectedly, writing
+to the pipe will raise a SIGPIPE signal. A SIGPIPE signal will cause the process to
+terminate unless it is handled or ignored. **s2n-tls does NOT handle SIGPIPE.** It is the
+responsibility of the application using s2n-tls to either handle or ignore SIGPIPE.
```

Comment:
I think we can say this shorter and clearer, like above. We shouldn't have to go super deep into how pipes work, we just need to remind people that signals exist and need to be ignored/handled.